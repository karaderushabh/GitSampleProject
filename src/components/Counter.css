.center {
  margin: auto;
  width: 30%;
  border: 3px solid #004953;
  padding: 10px;
}

input{
  width: 60px
}

.button {
  background-color: white;
  color: black;
  padding: 10px 10px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 8px 6px;
}

.buttonReset {
  border-radius: 12px;
  background-color: black;
  color: white;
  cursor: "pointer"
}

.button1 {
  border-radius: 12px;
  border: 2px solid #4CAF50;
  cursor: "pointer"

}

.pad {

  padding: 5%;
}

.button1:hover {
  background-color: #4CAF50;
  color: white;
  cursor: "pointer"

}

.button2 {
  border-radius: 12px;
  border: 2px solid red;
}

.button2:hover {
  background-color: red;
  color: white;
}

.button3 {
  background-color: white;
  color: black;
  border: 2px solid #e7e7e7;
}

.button3:hover {background-color: #e7e7e7;}

.p3 {
  font-family: "Lucida Console", "Courier New", monospace;
  width: 250px; 
}

.p1{
  font-family: sans-serif;
  padding: 8px 16px;
  text-align: center;
  background-color: #a8faf8;
  border: 2px solid #438684;
  width: 35%;
}

.p2{
  font-family: "Trirong", serif,monospace;
  background-color: rgba(129, 124, 124, 0);

}

.p4{
  border: 2px solid #438684;
  border-radius: 6px;
  width: 60%; 
  width: 280px;
}

.list {
  background-color: white;
  border-style: inset;
  color: black;
  border: 2px solid #e7e7e7;
  width: 350px;
}