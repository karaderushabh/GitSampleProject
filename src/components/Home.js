import React from "react";
import { BrowserRouter } from "react-router-dom";

function Home() {
  return (
    <>
      <h1>This is home page for counter</h1>
    </>
  );
}

export default Home;
