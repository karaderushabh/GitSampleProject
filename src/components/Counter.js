import { render } from "@testing-library/react";
import React from "react";
import { ReactDOM } from "react";
import "./Counter.css";

class Counter extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      counter: 0,
      customInput: 1,
      history: [],
    };
  }

  handleIncrement = async () => {
    console.log("increment function");

    console.log(JSON.stringify(this.state.history));
    await this.setState((prevState) => ({
      counter: prevState.counter + this.state.customInput,
      customInput: 1,
      history: [
        ...prevState.history,
        "Increment value by 1, Counter Value:  " +
          (prevState.counter + this.state.customInput),
      ],
    }));
    console.log(JSON.stringify(this.state.history));
  };

  handleDecrement = async () => {
    console.log("Decrement function");
    if (this.state.counter == 0) {
      alert("Counter cannot be negative");
    } else {
      await this.setState((prevState) => ({
        counter: prevState.counter - this.state.customInput,
        customInput: 1,
        history: [
          ...prevState.history,
          "Decrement value by 1, Counter Value:  " +
            (prevState.counter - this.state.customInput),
        ],
      }));
    }
  };

  customInputHandle = (e) => {
    this.setState({ customInput: parseInt(e.target.value) });
  };

  handleReset = () => {
    this.setState((prevState) => ({ counter: (prevState.counter = 0) }));
    this.setState(() => ({
      history: [],
    }));
  };

  render() {
    const renderHistory = this.state.history.map((item) => (
      <div className="list">
        <td>{item}</td>
      </div>
    ));

    return (
      <div className="center">
        <div align="center">
          <h1 className="p3">Counter App</h1>
          <h2 className="p1" data-testid="counter">
            Counter new Count : {this.state.counter}
          </h2>
          <button className="button button1" onClick={this.handleIncrement}>
            Increment
          </button>
          <button className="button button2" onClick={this.handleDecrement}>
            Decrement
          </button>
          <br />
          <br />
          <div className="p8"></div>
          <button className="button buttonReset" onClick={this.handleReset}>
            Reset
          </button>
          {renderHistory}
        </div>
        <div></div>
      </div>
    );
  }
}

export default Counter;
