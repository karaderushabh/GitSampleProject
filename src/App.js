import "./App.css";
import React from "react";
import Counter from "./components/Counter";
import { BrowserRouter, Route, Routes, Link } from "react-router-dom";
import Home from "./components/Home";

function App() {
  return (
    <>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Counter />} />
          <Route path="/Home" element={<Home />} />
        </Routes>
      </BrowserRouter>
    </>
  );
}

export default App;
