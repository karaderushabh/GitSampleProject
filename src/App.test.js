import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";
import Counter from "./components/Counter";
import { waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import React from "react";
import { ReactDOM } from "react";
import reportWebVitals from "./reportWebVitals";
import Home from "./components/Home";

test("inital counter App", () => {
  render(<Counter />);
  expect(screen.getByText("Counter App")).toBeInTheDocument();
});

test("Home Page", () => {
  render(<Home />);
  expect(screen.getByText("This is home page for counter")).toBeInTheDocument();
});

test("renders the counter with correct className and content", () => {
  const { container } = render(<Counter />);
  const counterElement = container.getElementsByClassName("p1")[0];
  expect(counterElement).toBeInTheDocument();

  // expect(counterElement).toHaveTextContent("Count : 0"); // Replace '0' with the expected counter value
});

test("Count visible", () => {
  const { container } = render(<Counter />);
  const counterElement = container.getElementsByClassName("p1")[0];
  expect(counterElement).toHaveTextContent("Count : 0");
});

test("increments the counter when the Increment button is clicked", async () => {
  const { getByText, getByTestId } = render(<Counter />);
  const initialCounter = getByTestId("counter").textContent;
  fireEvent.click(getByText("Increment"));
  await new Promise((resolve) => setTimeout(resolve, 0));
  const updatedCounter = getByTestId("counter").textContent;
  expect(parseInt(updatedCounter)).toBe(parseInt(initialCounter) + 1);
});

test("decrements the counter when the Decrement button is clicked", async () => {
  const { getByText, getByTestId } = render(<Counter />);
  const initialCounter = getByTestId("counter").textContent;
  fireEvent.click(getByText("Decrement"));
  await new Promise((resolve) => setTimeout(resolve, 0));
  const updatedCounter = getByTestId("counter").textContent;
  expect(parseInt(updatedCounter)).toBe(parseInt(initialCounter) - 1);
});

test("counter increments by 4 when + button is clicked four times", () => {
  render(<Counter />);
  const initialCounter = screen.getByTestId("counter").textContent;
  fireEvent.click(screen.getByText("Increment"));
  new Promise((resolve) => setTimeout(resolve, 0));
  const updatedCounter = screen.getByTestId("counter").textContent;
  expect(parseInt(updatedCounter)).toBe(parseInt(initialCounter) + 1);
  expect(parseInt(updatedCounter)).toBe(parseInt(initialCounter) + 1);
  expect(parseInt(updatedCounter)).toBe(parseInt(initialCounter) + 1);
  expect(parseInt(updatedCounter)).toBe(parseInt(initialCounter) + 1);
});

test("counter value should be 0 when + button is clicked 2 times and - 2 times", () => {
  render(<Counter />);
  fireEvent.click(screen.getByText("Increment"));
  fireEvent.click(screen.getByText("Increment"));
  fireEvent.click(screen.getByText("Decrement"));
  fireEvent.click(screen.getByText("Decrement"));

  const initialCounter = screen.getByTestId("counter").textContent;
  const updatedCounter = screen.getByTestId("counter").textContent;
  expect(parseInt(updatedCounter)).toBe(parseInt(initialCounter));
  // expect(screen.getByText("0")).toBeInTheDocument();
});

test("counter value should be present", () => {
  render(<Counter />);
  fireEvent.click(screen.getByText("Reset"));

  expect(screen.getByTestId("counter")).toBeInTheDocument();
});

test("renders the App component without crashing", () => {
  // Render your component
  render(
    <React.StrictMode>
      <App />
    </React.StrictMode>
  );

  // Check if a text from your App component is present
  const appText = screen.getByText(/Counter App/i); // Replace with the actual text in your App component
  // Assert that the text is present
  expect(appText).toBeInTheDocument();
});

jest.mock("./reportWebVitals", () => ({
  __esModule: true,
  default: jest.fn(),
}));

// it("renders correctly with a counter 1", () => {
//   const { asFragment } = render(<Counter />);
//   expect(asFragment()).toMatchSnapshot();
// });

jest.mock("web-vitals", () => ({
  __esModule: true,
  getCLS: jest.fn(),
  getFID: jest.fn(),
  getFCP: jest.fn(),
  getLCP: jest.fn(),
  getTTFB: jest.fn(),
}));

test("customInputHandle updates state with parsed integer value", () => {
  const { container } = render(<Counter />);
  const counterElement = container.getElementsByClassName("p2")[0];
  expect(counterElement).toHaveTextContent("Custom Input:");
});

test("handleReset resets counter and clears history", () => {
  const initialState = {
    counter: 5,
    customInput: 3,
    history: [
      "Increment value by 1, Counter Value: 6",
      "Decrement value by 1, Counter Value: 5",
    ],
  };

  const { container } = render(<Counter />);
  const inputElement = container.getElementsByClassName("p2")[0];

  // Trigger handleReset
  fireEvent.click(screen.getByText("Reset"));

  // Check if counter is reset
  expect(screen.getByText("Count : 0")).toBeInTheDocument();
});
