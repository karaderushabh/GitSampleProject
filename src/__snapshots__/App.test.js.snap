// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a counter 1`] = `
<DocumentFragment>
  <div
    class="center"
  >
    <div
      align="center"
    >
      <h1
        class="p3"
      >
        Counter App
      </h1>
      <h2
        class="p1"
        data-testid="counter"
      >
        Count : 0
      </h2>
      <button
        class="button button1"
      >
        Increment
      </button>
      <button
        class="button button2"
      >
        Decrement
      </button>
      <br />
      <br />
      <div
        class="p4"
      >
        <div
          class="pad"
        >
          <label
            class="p2"
          >
            Custom Input: 
          </label>
          <input
            type="number"
            value="1"
          />
        </div>
      </div>
      <button
        class="button buttonReset"
      >
        Reset
      </button>
    </div>
    <div />
  </div>
</DocumentFragment>
`;

exports[`renders correctly with a counter 1 1`] = `
<DocumentFragment>
  <div
    class="center"
  >
    <div
      align="center"
    >
      <h1
        class="p3"
      >
        Counter App
      </h1>
      <h2
        class="p1"
        data-testid="counter"
      >
        Count : 0
      </h2>
      <button
        class="button button1"
      >
        Increment
      </button>
      <button
        class="button button2"
      >
        Decrement
      </button>
      <br />
      <br />
      <div
        class="p4"
      >
        <div
          class="pad"
        >
          <label
            class="p2"
          >
            Custom Input: 
          </label>
          <input
            name="custom"
            type="number"
            value="1"
          />
        </div>
      </div>
      <button
        class="button buttonReset"
      >
        Reset
      </button>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
